cypress/integration/links.spec.js

```diff
@@ -0,0 +1,71 @@
+import { randHash } from '../utils/'
+
+const randUser = randHash()
+const fileName = 'empty.md'
+
+const getEditor = () => cy.get('#viewer #editor .ProseMirror')
+
+const clearContent = () => getEditor()
+	.type('{selectall}')
+	.type('{enter}')
+
+describe('test link marks', function() {
+	before(function() {
+		// Init user
+		cy.nextcloudCreateUser(randUser, 'password')
+		cy.login(randUser, 'password')
+
+		// Upload test files
+		cy.uploadFile('test.md', 'text/markdown')
+		cy.uploadFile(fileName, 'text/markdown')
+	})
+
+	beforeEach(function() {
+		cy.login(randUser, 'password')
+
+		cy.visit('/apps/files', {
+			onBeforeLoad(win) {
+				cy.stub(win, 'open')
+					.as('winOpen')
+
+			},
+		}).as('page')
+
+		cy.openFile(fileName)
+	})
+
+	describe('autolink', function() {
+		beforeEach(clearContent)
+		it('with protocol', () => {
+			cy.getFile('test.md')
+				.then($el => {
+					const id = $el.data('id')
+
+					const link = `${Cypress.env('baseUrl')}/file-name?fileId=${id}`
+					getEditor()
+						.type(link)
+
+					getEditor()
+						.get(`a[href*="${Cypress.env('baseUrl')}`)
+						.click({ force: true })
+
+					cy.get('@winOpen')
+						.should('have.been.calledOnce')
+						.should('have.been.calledWith', `/index.php/f/${id}`)
```

Comment:
This fails if NC is accessible with a sub path as the call will be like `/sub/path/index.php/f/${id}`. Not a big deal but this test may fail in the future if we change how NC is deployed in the CI job.

We could either check that the target of `@winOpen` ends with `/index.php/f/${id}` or get the sub path and include it in the expected value.